amethyst_tiles/src/map.rs

```diff
@@ -290,13 +301,36 @@ fn create_transform(map_dimensions: &Vector3<u32>, tile_dimensions: &Vector3<u32
 }
 
 #[allow(clippy::cast_precision_loss)]
-fn to_world(transform: &Matrix4<f32>, coord: &Point3<u32>) -> Vector3<f32> {
+fn to_world(
+    transform: &Matrix4<f32>,
+    coord: &Point3<u32>,
+    map_transform: Option<&Transform>,
+) -> Vector3<f32> {
     let coord_f = Point3::new(coord.x as f32, -1.0 * coord.y as f32, coord.z as f32);
-    transform.transform_point(&coord_f).coords
+    if let Some(map_trans) = map_transform {
+        map_trans
+            .global_view_matrix()
+            .try_inverse()
+            .unwrap()
+            .transform_point(&transform.transform_point(&coord_f))
+            .coords
+    } else {
+        transform.transform_point(&coord_f).coords
+    }
 }
 
-fn to_tile(transform: &Matrix4<f32>, coord: &Vector3<f32>) -> Option<Point3<u32>> {
-    let point = Point3::from(*coord);
+fn to_tile(
+    transform: &Matrix4<f32>,
+    coord: &Vector3<f32>,
+    map_transform: Option<&Transform>,
+) -> Option<Point3<u32>> {
+    let point = if let Some(map_trans) = map_transform {
+        map_trans
+            .global_view_matrix()
```

Comment:
see previous about `global_matrix`

Reply:
This one should be inversed though, so it's only doing one inverse